springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/MessageHelper.java

```diff
@@ -13,6 +19,10 @@ public class MessageHelper {
     private static final String ONE_OF = "oneOf";
 
     private static final Comparator<Message> byMessageName = Comparator.comparing(Message::getName);
+
+    // TODO Why do we need a SortedSet here? Using a comparator with only the message name will break deep equals on the Set
```

Comment:
I changed it to a HashSet to get back the ability to compare the sets with deep equals.  I'm not so sure what the original intention was and if this is really correct, but I would try not to hand out a TreeSet (especially if the type is erased to object ...).

Reply:
I believe the intention was constant ordering in the final json spec. This simplifies testing and adds a natural order. Could probably also be achieved on the level of the object mapper

Reply:
Yeah I remember now, if thats the only reason then this solution would work too